### DISCUSSION TITLE
更新日志 - 20230203

### DISCUSSION THREAD

**Discussion by jjm2473** [istoreos/istoreos#573] [Announcements]
### 系统：
1. 修复启动时 PARTUUID 冲突可能导致 overlay 挂载失败的问题
2. 修复 PARTUUID 冲突可能导致 boot 分区挂载失败或者固件升级时写错 boot 分区的问题 #535
3. 修复 lua 升级可能导致 argon 主题异常崩溃的问题

### 设备：
1. x86：修复某些设备在 GRUB 界面不自动启动 iStoreOS 的问题（倒计时失效） #558
2. R2S：固件支持 R2C 的网卡  #309 #358 #439

### 应用：
1. QuickStart：修复多个 RAID 功能相关bug  https://github.com/linkease/nas-packages-luci/issues/54
2. QuickStart：修复 docker 根目录状态的判断
4. QuickStart：修复 x86 安装器安装以后 GPT 分区表可能无效的问题 #556 #557
6. QuickStart：修复 efi 固件分区格式化系统盘剩余空间可能失败的问题 #408
7. QuickStart：x86 安装器跳过小于 2.5GiB 的硬盘
8. QuickStart：提高 lean's LEDE 固件的兼容性 https://github.com/linkease/nas-packages-luci/issues/55

**Comment by CoolAlien520** [comment]
能否加上无线功能

**Reply by jjm2473** [reply]
暂无计划支持wifi，有需要可以自己编译固件或者ipk

**Reply by wykl16399** [reply]
istoreos啥时候支持一下N1和玩客云？

**Reply by jjm2473** [reply]
除非openwrt官方内核支持，不然我们不会移植的

**Comment by CoderK7** [comment]
R68S更新最新固件，2.5G网卡跑不满千兆外网

**Reply by jjm2473** [reply]
不知道瓶颈在哪，可能要等内核升级以后才能改善

**Reply by CoderK7** [reply]
内核升级有计划嘛？另外目前还有断流的现象发生。

**Reply by jjm2473** [reply]
没有计划。如果之后openwrt官方也支持了3568，那就很快可以升级，不然就要看什么时候有空研究新内核了

**Reply by jjm2473** [reply]
20230310 固件应该可以跑满千兆外网了。实际上这跟2.5G网卡无关，而是千兆WAN口的CPU调度问题。

**Comment by wxchil** [comment]
我看很多教程在IPV6的设置上都有个"有状态+无状态",咱们这个固件怎么没有这条呢?
![image](https://user-images.githubusercontent.com/44988562/217129812-4b4b98ff-ecba-4c76-a42f-5530d29b4c09.png)

**Reply by jjm2473** [reply]
在RA设置里，SLACC就表示无状态

**Reply by jjm2473** [reply]
老版本的luci才是显示成你说的那样

**Reply by wxchil** [reply]
谢谢～

**Comment by popalanluck** [comment]
R68S刷最新版istoreos，内核4.19，安装shellclash后,无法正常使用tproxy模式。
如果改刷lean固件，同样4.19内核，tproxy模式完全正常。
![r68s-istoreos-tproxy-error](https://user-images.githubusercontent.com/81753503/217459125-10b3e3e2-1b7c-42f7-9e20-7f2359bdc762.png)

**Reply by jjm2473** [reply]
怎么个不正常法，有没有测试过passwall的tproxy

**Comment by toplezs** [comment]
Hope u can add English Language

**Comment by DHDAXCW** [comment]
R2S和 R2C的wan使用不同网卡，不要使用相同的delay值一起搞混，会引起兼容识别

**Reply by jjm2473** [reply]
具体有什么表现，会出现网卡不识别或者丢包吗？如果影响比较大，我打算根据phy_id动态配置delay。

**Reply by DHDAXCW** [reply]
我没有更多时间去测试。我之前调试有GMAC以太网换芯片，调试同样delay的值遇到以下几个问题：
原因是tx或rx通路某条通路不通，tx通路异常导致数据发不出去，同理rx通路异常导致接收不到数据。
这可能会导致几个问题：传输波动，以太网网络不通，丢包严重，测速不满。

临时解决方法：
tx rx delay的取值范围，如是[0x00, 0x7f]，使用rk提供delay扫描出结果。GMAC tx rx delay动态调整补丁V2.0
一般sdk kernel默认就有
每调整一次，测试下效果，当有多个值符合要求时，取平均值使用。
找到**合理的**tx rx delay后，替换掉dts中的值。
**ps**：如果tx rx delay的取值范围太大，那只能分开了

**Reply by jjm2473** [reply]
我看过这个自动扫描delay值的补丁，对于r2s,r2c应该用不上，他们两个本来delay值就相差无几，如果r2c没出现异常的丢包，应该就没问题。
这是其他固件使用的delay值：
```
		rtl8211e {
			phy_id = <0x001cc915>;
			rx_delay = <0x18>;
			tx_delay = <0x24>;
		};
		yt8521s {
			phy_id = <0x0000011a>;
			rx_delay = <0x12>;
			tx_delay = <0x22>;
		};
```

**Reply by DHDAXCW** [reply]
传输稳定吗？
试一下 iperf3 server----wan-r2c-lan----客户端电脑

**Reply by jjm2473** [reply]
我没有机器测试，用户反馈r2c外网测速有五六百Mbps

**Reply by jjm2473** [reply]
https://github.com/istoreos/istoreos/issues/358

**Reply by DHDAXCW** [reply]
这特太勉强了。。。就这样吧

**Reply by jjm2473** [reply]
r2s是openwrt官方支持的，不是使用sdk内核，我改的dts只是为了不写死phy的id，自动识别r2c的phy
